Description: don't throw exceptions from EvaluatorScriptClass
Origin: upstream, http://code.qt.io/cgit/qt-labs/qbs.git/commit/?id=6bf4836f381ece41
Bug: https://bugreports.qt.io/browse/QBS-863
Last-Update: 2015-08-26

--- a/src/lib/corelib/language/evaluatorscriptclass.cpp
+++ b/src/lib/corelib/language/evaluatorscriptclass.cpp
@@ -373,7 +373,11 @@
 
     for (ValuePtr next = value; next; next = next->next()) {
         QScriptValue v = data->evaluator->property(next->definingItem(), propertyName);
-        data->evaluator->handleEvaluationError(next->definingItem(), propertyName, v);
+        ScriptEngine * const se = static_cast<ScriptEngine *>(engine());
+        if (se->hasErrorOrException(v)) {
+            *result = se->lastErrorValue(v);
+            return;
+        }
         if (v.isUndefined())
             continue;
         lst << v;
@@ -384,10 +388,7 @@
     quint32 k = 0;
     for (int i = 0; i < lst.count(); ++i) {
         const QScriptValue &v = lst.at(i);
-        if (v.isError()) {
-            *result = v;
-            return;
-        }
+        QBS_ASSERT(!v.isError(), continue);
         if (v.isArray()) {
             const quint32 vlen = v.property(QStringLiteral("length")).toInt32();
             for (quint32 j = 0; j < vlen; ++j)
